source/blog/2018-02-12-ember-3-0-released.md

```diff
@@ -0,0 +1,250 @@
+
+- [`Ember.Binding`](/deprecations/v2.x/#toc_ember-binding)
+
+#### Legacy {{render}} helper
+
+Ember’s `{{render}}` helper presaged the idea of component-driven rendering architecture as we now know it. Despite this, it was also an API coupled to Ember’s `Controller` system, was subject to edge cases where the context of a template was difficult to discern, and additionally was entangled with Ember’s routing system. Through the 2.x cycle we’ve worked to ensure each use-case for `{{render}}` has a viable solution in Ember’s component-driven system.
+
+Fully deprecated since Ember 2.11, in 3.0 the `{{render}}` helper will be removed. Please see the deprecation guide for more details:
+
+- [`{{#render}}` helper with a block](/deprecations/v2.x/#toc_render-helper-with-block)
+- [Model param in `{{render}}` helper](/deprecations/v2.x/#toc_model-param-in-code-render-code-helper)
+- [`{{render}}` helper](/deprecations/v2.x/#toc_code-render-code-helper)
+- [Rendering into a `{{render}}` helper that resolves to an `{{outlet}}`](/deprecations/v2.x/#toc_rendering-into-a-render-helper-that-resolves-to-an-outlet)
+
+#### Legacy Ember.K utility
+
+Ember has historically provided a short-to-type noop function for use as a stub on parent classes or in other situations where a noop might be useful. With the advent of ECMAScript 2015 [method definitions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions) and [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), this kind of utility has become more confusing than convenient.
+
+In Ember 3.0 the `Ember.K` function will be removed. For more details, please see the deprecation guide:
+
+- [`Ember.K`](/deprecations/v2.x/#toc_code-ember-k-code)
+
+#### Consuming Ember through Bower
+
+Ember.js 3.0 and later releases will no longer be published to bower. If you still use bower today, we strongly suggest you begin the migration of your dependencies to npm.
+
+<hr>
+
+In the past, we have failed to make clear what this means for users of globals mode. We apologize for this lack of clarity.
+
+In short, globals builds of any kind will no longer be maintained or published. Not only will they not be availabe in bower, but you
+will likely not find them on CDNs or by download from the emberjs.com website. While the initial release of Ember 3.0 does contain
+a globals build of bower as part of the ember-source npm package, this too will disappear before long. What this means is,
+if you need to continue using globals mode in Ember 3.0, as a part of your build process, you will need to compile a build of Ember itself
+using ember-cli. We know this will cause considerable hardship to users of globals mode, but this is important in order to be able to make
+progress on some long awaited features such as splitting Ember into smaller packages and tree shaking.
```

Comment:
Above from previous comment should be given close attention.

Remainder below is unaltered.

Reply:
@Gaurav0 I would personally think the handwriting has been on the wall for globals mode for a while, but that may be because I've been relatively close to things. Maybe we could do something along these lines?

> Continuing our shift away from bundling Ember as one large Javascript file, which we have been aiming for since the introduction of Ember-CLI, we’re now officially ending the publishing of new Ember versions on Bower or CDNs.
> 
> We recognize that for a small part of our community, this may require reworking build processes (it will entail additional work for Ember Twiddle as well). We're sorry about that, but feel unlocking long-awaited features such as tree-shaking and allowing publishing Ember as smaller packages makes this a worthwhile cost to pay as a community.

The above probably needs further wordsmithing, but I think conveys the goals behind these changes ... @rwjblue @wycats @locks any thoughts on the above? I wondered about talking about the goal of allowing easier npm access in other build systems, but wasn't sure if that'd be promising too much ...

Reply:
Hi, @acorncom. Thanks for working on the wordsmithing. I appreciate the feedback and I'm incorporating it. I'm also going to make the file rename to correct the date a separate commit.

Reply:
I would personally think the handwriting has been on the wall for globals mode for a while

Perhaps, which is why I am encouraging us to deprecate it explicitly, and soon. It is important to communicate clearly and explicitly when something is going away, as soon as we know it is, and then give people lots of time to migrate.

My concern that brought up this PR is that I didn't translate automatically from "publishing to bower is going away" to "all builds (globals or otherwise) are going away and I need to compile Ember myself using ember-cli". In fact, several other core team members I spoke to weren't aware of this either.

Similarly, if using globals mode now is not recommended because it might be going away, well that is kinda the definition of the word deprecated. People get upset when they aren't warned, so let's at least avoid the next mistake if we can't fix this one.